geode-connectors/src/acceptanceTest/java/org/apache/geode/connectors/jdbc/JdbcDistributedTest.java

```diff
@@ -445,6 +446,36 @@ public void getReadsFromDB() throws Exception {
     });
   }
 
+  @Test
+  public void getReadsFromDBWithCompositeKey() throws Exception {
+    createTable();
+    createRegionUsingGfsh();
+    createJdbcDataSource();
+    createMapping(REGION_NAME, DATA_SOURCE_NAME, Employee.class.getName(), true, "id,age");
+    server.invoke(() -> {
+      PdxInstance pdxEmployee1 =
+          ClusterStartupRule.getCache().createPdxInstanceFactory(Employee.class.getName())
+              .writeString("id", "id1").writeString("name", "Emp1").writeInt("age", 55).create();
+      JSONObject compositeKey1 = new JSONObject();
+      compositeKey1.put("id", pdxEmployee1.getField("id"));
+      compositeKey1.put("age", pdxEmployee1.getField("age"));
+      String key = compositeKey1.toString();
+      Region<Object, Object> region = ClusterStartupRule.getCache().getRegion(REGION_NAME);
+      region.put(key, pdxEmployee1);
+      region.invalidate(key);
+      JdbcWriter<Object, Object> writer =
+          (JdbcWriter<Object, Object>) region.getAttributes().getCacheWriter();
+      long writeCallsCompletedBeforeGet = writer.getTotalEvents();
+
+      Employee result = (Employee) region.get(key);
+
+      assertThat(result.getId()).isEqualTo("id1");
+      assertThat(result.getName()).isEqualTo("Emp1");
+      assertThat(result.getAge()).isEqualTo(55);
+      assertThat(writer.getTotalEvents()).isEqualTo(writeCallsCompletedBeforeGet);
```

Comment:
We don't need this check. As we know, the writer should not be getting invoked for a get, which is testing geode behavior. As test says, we should have added one more record with different ids value and verified only the expected one is returned.

Reply:
We just copied this test from another which also had the same logic of testing to make sure the writer was not called. I agree with you but thought we might be missing something and it does not hurt to do this check.